src/transaction.ts

```diff
@@ -0,0 +1,215 @@
+import { CoinSelector } from './coinselection/coinSelector';
+import {
+  UtxoInterface,
+  ChangeAddressFromAssetGetter,
+  RecipientInterface,
+} from './types';
+import { Psbt, networks, address as laddress } from 'liquidjs-lib';
+import { estimateTxSize } from './wallet';
+
+export interface BuildTxArgs {
+  psetBase64: string;
+  unspents: UtxoInterface[];
+  recipients: RecipientInterface[];
+  coinSelector: CoinSelector;
+  changeAddressByAsset: ChangeAddressFromAssetGetter;
+  feeAssetHash: string;
+  addFee?: boolean;
+  satsPerByte?: number;
+}
+
+function validateAndProcess(args: BuildTxArgs): BuildTxArgs {
```

Comment:
```suggestion
function validateBuildTxArgs(args: BuildTxArgs): BuildTxArgs {
```

At the moment this is a hybrid of sanitizing and validating... I would go for the validation, so this would return void and only raise exception in case of bad arguments.